pkg/interface/src/apps/chat/components/lib/chat-input.js

```diff
@@ -339,96 +179,54 @@ export class ChatInput extends Component {
-      .defaultTo([])
-      .map('author')
-      .uniq()
-      .reverse()
-      .value();
-
-    const codeTheme = state.code ? ' code' : '';
-
-    const options = {
-      mode: MARKDOWN_CONFIG,
-      theme: 'tlon' + codeTheme,
-      lineNumbers: false,
-      lineWrapping: true,
-      scrollbarStyle: 'native',
-      cursorHeight: 0.85,
-      placeholder: state.code ? 'Code...' : props.placeholder,
-      extraKeys: {
-        Tab: cm =>
-          this.patpAutocomplete(cm.getValue(), true),
-        'Enter': () => {
-          this.messageSubmit();
-          if (this.state.code) {
-            this.toggleCode();
-          }
-        },
-        'Shift-3': cm =>
-          cm.getValue().length === 0
-            ? this.toggleCode()
-            : CodeMirror.Pass
-      }
-    };
-
     return (
-      <div className="chat pa3 cf flex black white-d bt b--gray4 b--gray1-d bg-white bg-gray0-d relative"
-      style={{ flexGrow: 1 }}
-      >
-        <ShipSearch
-          popover
-          onSelect={this.completePatp}
-          onClear={this.clearSearch}
-          contacts={props.contacts}
-          candidates={candidates}
-          searchTerm={this.state.patpSearch}
-          cm={this.editor}
-        />
-        <div
-          className="fl"
-          style={{
-            marginTop: 6,
-            flexBasis: 24,
-            height: 24
-          }}
-        >
-        {img}
+      <div className={
+             "pa3 cf flex black white-d bt b--gray4 b--gray1-d bg-white" +
+             "bg-gray0-d relative"
+           }
+           style={{ flexGrow: 1 }}>
+        <div className="fl"
+             style={{
+               marginTop: 6,
+               flexBasis: 24,
+               height: 24
+             }}>
+          {avatar}
         </div>
-        <div
-          className="fr h-100 flex bg-gray0-d lh-copy pl2 w-100 items-center"
-          style={{ flexGrow: 1, maxHeight: '224px', width: 'calc(100% - 72px)' }}
-        >
-          <CodeEditor
-            value={this.props.message}
-            options={options}
-            editorDidMount={(editor) => {
-            this.editor = editor;
-            if (!/Android|webOS|iPhone|iPad|iPod|BlackBerry/i.test(
-                navigator.userAgent
-              )) {
-              editor.focus();
-              }
-            }}
-            onChange={(e, d, v) => this.messageChange(e, d, v)}
-          />
-        </div>
-        <div className="ml2 mr2"
-          style={{ height: '16px', width: '16px', flexBasis: 16, marginTop: 10 }}>
+        <ChatEditor
+          inCodeMode={state.inCodeMode}
+          submit={this.submit}
+          onUnmount={props.onUnmount}
+          message={props.message}
+          placeholder='Message...' />
+        <div className="ml3 mr3"
+             style={{
+               height: '16px',
+               width: '16px',
+               flexBasis: 16,
+               marginTop: 10
+             }}>
```

Comment:
This is a little off.

Before (livenet right now):

<img width="207" alt="Screen Shot 2020-08-03 at 7 37 58 PM" src="https://user-images.githubusercontent.com/20846414/89237368-00614300-d5c1-11ea-8f92-fb1b36322442.png">

After (localhost:9000):

<img width="204" alt="Screen Shot 2020-08-03 at 7 37 47 PM" src="https://user-images.githubusercontent.com/20846414/89237385-06572400-d5c1-11ea-9f2a-26629c10e719.png">

Reply:
Reverting to ml2 and mr2